rasa/telemetry.py

```diff
@@ -0,0 +1,506 @@
+import asyncio
+import datetime
+import functools
+from functools import wraps
+import hashlib
+import json
+import logging
+import multiprocessing
+import os
+from pathlib import Path
+import platform
+from subprocess import CalledProcessError, STDOUT, check_output  # skipcq:BAN-B404
+import sys
+import textwrap
+from typing import Any, Dict, Optional, Text
+import uuid
+
+import aiohttp
+from terminaltables import SingleTable
+
+import rasa
+from rasa.constants import (
+    CONFIG_FILE_TELEMETRY_KEY,
+    CONFIG_TELEMETRY_DATE,
+    CONFIG_TELEMETRY_ENABLED,
+    CONFIG_TELEMETRY_ID,
+    DOCS_URL_TELEMETRY,
+)
+from rasa.importers.importer import TrainingDataImporter
+from rasa.utils import common as rasa_utils
+
+logger = logging.getLogger(__name__)
+
+SEGMENT_ENDPOINT = "https://api.segment.io/v1/track"
+
+TELEMETRY_ENVIRONMENT_VARIABLE = "RASA_TELEMETRY_ENABLED"
+TELEMETRY_ENVIRONMENT_DEBUG_VARIABLE = "RASA_TELEMETRY_DEBUG"
+
+TELEMETRY_HTTP_TIMEOUT = 2  # Seconds
+TELEMETRY_ID = "metrics_id"
+TELEMETRY_ENABLED_BY_DEFAULT = True
+
+# if one of these environment variables is set, we assume to be running in CI env
+CI_ENVIRONMENT_TELL = [
+    "bamboo.buildKey",
+    "BUILD_ID",
+    "BUILD_NUMBER",
+    "BUILDKITE",
+    "CI",
+    "CIRCLECI",
+    "CONTINUOUS_INTEGRATION",
+    "GITHUB_ACTIONS",
+    "HUDSON_URL",
+    "JENKINS_URL",
+    "TEAMCITY_VERSION",
+    "TRAVIS",
+]
+
+# If updating or creating a new event, remember to update
+# https://rasa.com/docs/rasa/telemetry
```

Comment:
WDYT of doing an approach using a JSON file (`events.json`) like we do for Rasa X?

Reply:
I'd say yml is easier to write, but yes seems like a good idea.